src/ripple/basics/feeunits.h

```diff
@@ -0,0 +1,431 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#ifndef BASICS_FEES_H_INCLUDED
+#define BASICS_FEES_H_INCLUDED
+
+#include <ripple/basics/BasicConfig.h>
+#include <ripple/basics/contract.h>
+#include <ripple/basics/safe_cast.h>
+#include <ripple/beast/cxx17/type_traits.h>
+#include <ripple/json/json_value.h>
+#include <boost/multiprecision/cpp_int.hpp>
+#include <boost/units/base_dimension.hpp>
+#include <boost/units/base_unit.hpp>
+#include <boost/units/dimensionless_quantity.hpp>
+/* The library provides I/O functions for quantities, but
+    always include units. We do not want units when we're
+    converting these types for API consumption, so do NOT
+    include io.hpp.
+*/
+//#include <boost/units/io.hpp>
+#include <boost/units/make_system.hpp>
+#include <boost/units/operators.hpp>
+#include <boost/units/quantity.hpp>
+#include <boost/units/static_constant.hpp>
+#include <boost/units/unit.hpp>
+#include <limits>
+#include <utility>
+
+#include <cassert>
+#include <cmath>
+#include <string>
+#include <iosfwd>
+#include <ios>
+#include <sstream>
+
+namespace boost {
+namespace units {
+
+// Allow quantities to be output as just their numeric value.
+// This will make API interface work easier.
+template<class Char, class Traits, class Unit, class T>
+std::basic_ostream<Char, Traits>&
+operator<<(std::basic_ostream<Char, Traits>& os,
+    const quantity<Unit, T>& q)
+{
+    os << q.value();
+
+    return(os);
+}
+
+// get string representation of a quantity
```

Comment:
Even better, I can just do `return to_string(t.value());`. I originally used `operator<<` to future-proof it in case I only wanted to change the output format in the future. I doubt that's going to happen, though, honestly.